application/src/main/java/org/togetherjava/tjbot/commands/moderation/KickCommand.java

```diff
@@ -0,0 +1,131 @@
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either kick other users or
+ * to kick the specific given user (for example a moderator attempting to kick an admin).
+ */
+public final class KickCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(KickCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "kick";
+    private static final String ACTION_VERB = "kick";
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Constructs an instance.
+     */
+    public KickCommand() {
+        super(COMMAND_NAME, "Kicks the given user from the server", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, TARGET_OPTION, "The user who you want to kick", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be kicked", true);
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getSoftModerationRolePattern())
+            .asMatchPredicate();
+    }
+
+    private static void handleAbsentTarget(@NotNull Interaction event) {
+        event.reply("I can not kick the given user since they are not part of the guild anymore.")
+            .setEphemeral(true)
+            .queue();
+    }
+
+    private static void kickUser(@NotNull Member target, @NotNull Member author,
+            @NotNull String reason, @NotNull Guild guild, @NotNull SlashCommandEvent event) {
+        User targetUser = target.getUser();
+        event.getJDA()
+            .openPrivateChannelById(targetUser.getId())
+            .flatMap(channel -> channel.sendMessage(
+                    """
+                            Hey there, sorry to tell you but unfortunately you have been kicked from the server %s.
+                            If you think this was a mistake, please contact a moderator or admin of the server.
+                            The reason for the kick is: %s
+                            """
+                        .formatted(guild.getName(), reason)))
+            .mapToResult()
+            .flatMap(sendDmResult -> {
+                logger.info("'{}' ({}) kicked the user '{}' ({}) from guild '{}' for reason '{}'.",
+                        author.getUser().getAsTag(), author.getId(), targetUser.getAsTag(),
+                        targetUser.getId(), guild.getName(), reason);
+
+                return guild.kick(target, reason)
+                    .reason(reason)
+                    .map(kickResult -> sendDmResult.isSuccess());
+            })
+            .map(hasSentDm -> {
+                String dmNotice =
+                        Boolean.TRUE.equals(hasSentDm) ? "" : "(Unable to send them a DM.)";
+                return ModerationUtils.createActionResponse(author.getUser(),
+                        ModerationUtils.Action.KICK, targetUser, dmNotice, reason);
```

Comment:
I find this to look really weird and ugly.

The ternarny makes it look confusing, and the name of the variable makes it even worse.
What's dmNotice? Magic String you could say.
I'd rather have specific methods for ActionResponses with a "dmWasSuccessfull" boolean. (not this name, but you get the point)

This'd make it a lot clearer imo.

Reply:
what is the `Boolean.TRUE.equals` for? just do `hasSentDm ? ...`

Reply:
* @Tais993 agreed, reworked in the next commit (with individual methods)
* @java-coding-prodigy this is the idiomatic way to write `foo != null && foo`, i.e. null-safe `true` check.

Reply:
done.

Reply:
oh that might be null forgot that